R/zchunk_L1233.Elec_water.R

```diff
@@ -39,89 +38,301 @@ module_water_L1233.Elec_water_DISABLED <- function(command, ...) {
+
+    # ADD TECHNOLOGIES WITH NO COOLING FOR ALL HISTORICAL YEARS
+    L1233.weights_ctry_elec_F_Yh_cool %>%
+      filter(plant_type == "no cooling") %>%
+      mutate(value = 1) %>%
+      select(-year) %>%
+      repeat_add_columns(tibble(year = HISTORICAL_YEARS)) %>%
+      bind_rows(L1233.weights_ctry_elec_F_Yh_cool) %>%
+      filter(year %in% HISTORICAL_YEARS) %>%
+      rename(share = value) -> L1233.weights_ctry_elec_F_Yh_cool_withNoCool
+
+
+    # STEP 2. MULTIPLY SHARES BY COUNTRY LEVEL GENERATION TO GET ELECTRICITY...
+    # ... OUTPUT BY GENERATING AND COOLING TECHNOLOGIES
+    L1233.weights_ctry_elec_F_Yh_cool_withNoCool %>%
+      left_join(L1233.out_EJ_ctry_elec_F_Yh, by = c("iso", "year", "fuel", "sector")) %>%
+      rename(gen = value) %>%
+      mutate(output = share * gen) %>%
+      select(-share, -gen) %>%
+      mutate(output = if_else(is.na(output) == T, 0, output)) -> L1233.output_ctry_elec_F_Yh_cool
+
+    # STEP 3: AGGREGATE COUNTRY-LEVEL DATA TO NEW GCAM REGIONS
+    # (a) ...WITH COOLING TECHNOLOGIES
+    L1233.output_ctry_elec_F_Yh_cool %>%
+      left_join_error_no_match(select(iso_GCAM_regID,-country_name, -region_GCAM3), by = "iso") %>%
+      group_by(GCAM_region_ID, sector, fuel, technology, cooling_system, water_type, year) %>%
+      summarise(output = sum(output)) %>%
+      ungroup() -> L1233.weights_R_elec_F_Yh_cool
+    # (b) ...WITHOUT COOLING TECHNOLOGIES
+    L1233.output_ctry_elec_F_Yh_cool %>%
+      left_join_error_no_match(select(iso_GCAM_regID,-country_name, -region_GCAM3), by = "iso") %>%
+      group_by(GCAM_region_ID, sector, fuel, technology, year) %>%
+      summarise(output = sum(output)) %>%
+      ungroup() -> L1233.weights_R_elec_F_Yh
+
+    # STEP 4: MATCH REGIONAL-LEVEL SHARES OF COOLING TECHNOLOGY WITH GENERATION TECHNOLOGY
+    elec_tech_water_map_cal %>%
+      select(sector, fuel, technology, cooling_system, water_type, plant_type) %>%
+      repeat_add_columns(unique(select(iso_GCAM_regID, GCAM_region_ID))) %>%
+      select(GCAM_region_ID, sector, fuel, technology, cooling_system, water_type, plant_type) %>%
+      left_join(L1233.weights_R_elec_F_Yh_cool,
+                by = c("GCAM_region_ID", "sector", "fuel", "technology", "cooling_system", "water_type")) %>%
+      rename(output_cool = output) %>%
+      left_join(L1233.weights_R_elec_F_Yh,
+                by = c("GCAM_region_ID", "sector", "fuel", "technology", "year")) %>%
+      mutate(share = output_cool / output) %>% select(-output_cool, -output) %>%
+      arrange(GCAM_region_ID, sector, fuel, technology, cooling_system, water_type, plant_type, year) %>%
+      mutate(share = if_else(year == tail(HISTORICAL_YEARS, 1) & is.na(share), -1, share)) ->
+      L1233.shares_R_elec_F_tech_Yh_cool
+
+    ## REMOVE NA VALUES BY PULLING FROM YEAR AHEAD IN EACH GROUP
+    for(i in (nrow(L1233.shares_R_elec_F_tech_Yh_cool) - 1) : 1){
```

Comment:
Agree, this is horrible. `lag `and `lead` won't cut it as there's dependency. It's more of a fill operation. I'll spend a bit more time on it to see if I can get rid of the for loop!